main_test.go

```diff
@@ -151,10 +159,27 @@ func newTest(t *testing.T) test {
 	return test{
 		config: controllertest.SetupEnv(t),
 		objects: []client.Object{
+			&corev1.Namespace{
```

Comment:
There's a bit of magic happening in the test; we create a specific number of "initial resources" which are created before the controller is started: https://github.com/snyk/kubernetes-scanner/blob/main/main_test.go#L80

By adding the namespace as the first resource, this will now be the only resource that exists at the beginning.
I'd suggest increasing that `initialResources` variable by one, or changing the order of the resources here.